workspace/resource_notebook.go

```diff
@@ -203,31 +209,29 @@ func (a NotebooksAPI) Delete(path string, recursive bool) error {
 	}, nil)
 }
 
+func setComputedProperties(d *schema.ResourceData, c *common.DatabricksClient) {
+	d.Set("url", c.FormatURL("#workspace", d.Id()))
+	d.Set("workspace_path", "/Workspace"+d.Id())
+}
+
 // ResourceNotebook manages notebooks
 func ResourceNotebook() common.Resource {
 	s := FileContentSchema(map[string]*schema.Schema{
 		"language": {
 			Type:     schema.TypeString,
 			Optional: true,
+			Computed: true, // we need it because it will be filled by the provider or backend
 			ValidateFunc: validation.StringInSlice([]string{
 				Scala,
 				Python,
 				R,
 				SQL,
 			}, false),
-			DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-				source := d.Get("source").(string)
-				if source == "" {
-					return false
-				}
-				ext := strings.ToLower(filepath.Ext(source))
-				return old == extMap[ext].Language
-			},
 		},
 		"format": {
 			Type:     schema.TypeString,
 			Optional: true,
```

Comment:
You don't need to change it in this PR, but is this really optional? From the docs, users can't specify this, and it seems like we always set it in Create based on the language/source rather than accepting the user-provided value.